stl/inc/xutility

```diff
@@ -2101,13 +2109,9 @@ namespace ranges {
 
             template <class _Ty>
             _NODISCARD static _CONSTEVAL _Choice_t<_St> _Choose() noexcept {
-                constexpr bool _Is_lvalue = is_lvalue_reference_v<_Ty>;
-
                 if constexpr (is_array_v<remove_reference_t<_Ty>>) {
-                    if constexpr (_Is_lvalue) {
-                        return {_St::_Array, true};
-                    }
-                } else if constexpr (_Is_lvalue && _Has_member<_Ty>) {
+                    return {_St::_Array, true};
+                } else if constexpr (_Has_member<_Ty>) {

```

Comment:
Changes applied.

Reply:
Thanks a lot. I read the paper multiple times before i thought about implementing it and then some more while writing it and now i have been staring at it for 10 minutes an got to the point where I have a sliver of an idea what you did.

Gotta love C++

Reply:
The short version is "after we check `is_lvalue || enable_safe` we just pretend every argument is an lvalue to avoid ever burning compile time on overload resolution for rvalues". Don't feel bad: it took me a while to convince the Ranges experts (including the author of the proposal and myself) that this was correct.

Reply:
The short version is "after we check `is_lvalue || enable_safe` we just pretend every argument is an lvalue to avoid ever burning compile time on overload resolution for rvalues". Don't feel bad: it took me a while to convince the Ranges experts (including the author of the proposal and myself) that this was correct.